src/ethereum/london/vm/__init__.py

```diff
@@ -82,10 +85,64 @@ class Evm:
     running: bool
     message: Message
     output: Bytes
-    accounts_to_delete: Dict[Address, Address]
+    accounts_to_delete: Set[Address]
+    touched_accounts: Set[Address]
     has_erred: bool
-    children: List["Evm"]
     return_data: Bytes
     error: Optional[Exception]
     accessed_addresses: Set[Address]
     accessed_storage_keys: Set[Tuple[Address, Bytes32]]
+
+
+def incorporate_child_evm_successful(evm: Evm, child_evm: Evm) -> None:
+    """
+    Incorporate the state of a successful `child_evm` into the parent `evm`.
+
+    Parameters
+    ----------
+    evm :
+        The parent `EVM`.
+    child_evm :
+        The child evm to incorporate.
+    """
+    evm.gas_left += child_evm.gas_left
+    evm.logs += child_evm.logs
+    evm.refund_counter += child_evm.refund_counter
+    evm.accounts_to_delete.update(child_evm.accounts_to_delete)
+    evm.touched_accounts.update(child_evm.touched_accounts)
+    if account_exists_and_is_empty(
```

Comment:
I have wrapped every touch with this `account_exists_and_is_empty()` check. This is not necessary, because the check has to be repeated at the end of the transaction anyway. I've included it to demonstrate that only touches of empty accounts need to be tracked, but this is arguably an optimization that shouldn't be included in the specs.